operator/pgdump/dump.go

```diff
@@ -81,6 +81,13 @@ func Dump(namespace string, clientset *kubernetes.Clientset, client *rest.RESTCl
 		}
 	}
 
+	// get the pgcluster CRD for cases where a CCPImagePrefix is specified
+	cluster := crv1.Pgcluster{}
+	_, err = kubeapi.Getpgcluster(client, &cluster, task.Spec.Parameters[config.LABEL_PG_CLUSTER], namespace)
+	if err != nil {
+		log.Error(err)
+	}
```

Comment:
There are a few things here in which if they are not set, this will totally fail.

Recommendation:

```go
clusterName := task.Spec.Parameters[config.LABEL_PG_CLUSTER]

if clusterName == "" {
    log.Error(err)
    return
}

if _, err := kubeapi.Getpgcluster(client, &cluster, clusterName, namespace); err != nil {
  log.Error(err)
  return
}
```

Reply:
Agreed, I'll implement that change.